apps/api/src/app/inbound-parse/usecases/inbound-email-parse/inbound-email-parse.usecase.ts

```diff
@@ -89,7 +89,11 @@ export class InboundEmailParse {
       selectEnvironment: 'apiKeys dns',
     });
 
-    const message = await this.messageRepository.findOne({ transactionId, _environmentId: environment._id });
+    const message = await this.messageRepository.findOne({
+      transactionId,
+      _environmentId: environment._id,
+      _subscriberId: subscriber._id,
```

Comment:
Added subscriberId so the query will fit the index in line 148 in message scheme file